chaintest/action_test_helpers.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+var _ state.Mutable = (*InMemoryStore)(nil)
+
+// InMemoryStore is a storage that acts as a wrapper around a map and implements state.Mutable.
+type InMemoryStore struct {
+	Storage map[string][]byte
+}
+
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{
+		Storage: make(map[string][]byte),
+	}
+}
```

Comment:
We could have modified the API such that these methods never return an error for instance because all of them cannot really fail, in the case of `GetValue` then the check of `nil` could have just been used for the value but it's a bad idea for an API that tries to mimic other databases so it "simulates" errors.